bigbluebutton-html5/imports/ui/components/audio/container.jsx

```diff
@@ -80,54 +77,44 @@ class AudioContainer extends PureComponent {
   }
 
   componentDidMount() {
-    const { meetingIsBreakout } = this.props;
+    const {
+      meetingIsBreakout,
+      openVideoPreviewModal,
+    } = this.props;
 
     this.init().then(() => {
       if (meetingIsBreakout && !Service.isUsingAudio()) {
-        this.joinAudio();
+        joinAudioAutomatically();
+        return;
       }
-    });
-  }
-
-  componentDidUpdate(prevProps) {
-    if (this.userIsReturningFromBreakoutRoom(prevProps)) {
-      this.joinAudio();
-    }
-  }
-
-  /**
-   * Helper function to determine wheter user is returning from breakout room
-   * to main room.
-   * @param  {[Object} prevProps prevProps param from componentDidUpdate
-   * @return {boolean}           True if user is returning from breakout room
-   *                             to main room. False, otherwise.
-   */
-  userIsReturningFromBreakoutRoom(prevProps) {
-    const { hasBreakoutRooms } = this.props;
-    const { hasBreakoutRooms: hadBreakoutRooms } = prevProps;
-    return hadBreakoutRooms && !hasBreakoutRooms;
-  }
-
-  /**
-   * Helper function that join (or not) user in audio. If user previously
-   * selected microphone, it will automatically join mic (without audio modal).
-   * If user previously selected listen only option in audio modal, then it will
-   * automatically join listen only.
-   */
-  joinAudio() {
-    if (Service.isConnected()) return;
-
-    const {
-      userSelectedMicrophone,
-      userSelectedListenOnly,
-    } = this.props;
 
-    if (userSelectedMicrophone) {
-      joinMicrophone(true);
-      return;
-    }
+      Breakouts.find().observeChanges({
+        removed() {
+          // if the user joined a breakout room, the main room's audio was
+          // programmatically dropped to avoid interference. On breakout end,
+          // offer to rejoin main room audio only if the user is not in audio
+          // already
+          const enableVideo = getFromUserSettings('bbb_enable_video',
+            KURENTO_CONFIG.enableVideo);
+          const autoShareWebcam = getFromUserSettings('bbb_auto_share_webcam',
+            KURENTO_CONFIG.autoShareWebcam);
+
+          if (Service.isUsingAudio()) {
+            if (enableVideo && autoShareWebcam) {
+              openVideoPreviewModal();
+            }
+
+            return;
+          }
 
-    if (userSelectedListenOnly) joinListenOnly();
+          joinAudioAutomatically().then(() => {
```

Comment:
Hi @mariogasparoni! Just one request. I think a `catch` block with a simple log message would be great here. Great PR by the way!